packages/pn-personafisica-webapp/src/components/Contacts/SpecialContacts.tsx

```diff
@@ -0,0 +1,321 @@
+import { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import AddIcon from '@mui/icons-material/Add';
+import { Card, CardContent, Divider, Stack, Typography } from '@mui/material';
+import { appStateActions, useIsMobile } from '@pagopa-pn/pn-commons';
+import { ButtonNaked } from '@pagopa/mui-italia';
+
+import { PFEventsType } from '../../models/PFEventsType';
+import {
+  AddressType,
+  ChannelType,
+  DigitalAddress,
+  SaveDigitalAddressParams,
+  Sender,
+} from '../../models/contacts';
+import { Party } from '../../models/party';
+import { createOrUpdateAddress, deleteAddress } from '../../redux/contact/actions';
+import { contactsSelectors } from '../../redux/contact/reducers';
+import { useAppDispatch, useAppSelector } from '../../redux/hooks';
+import PFEventStrategyFactory from '../../utility/MixpanelUtils/PFEventStrategyFactory';
+import { internationalPhonePrefix } from '../../utility/contacts.utility';
+import AddSpecialContactDialog from './AddSpecialContactDialog';
+import ContactCodeDialog from './ContactCodeDialog';
+import DeleteDialog from './DeleteDialog';
+import SpecialContactElem from './SpecialContactItem';
+
+enum ModalType {
+  EXISTING = 'existing',
+  CODE = 'code',
+  DELETE = 'delete',
+  SPECIAL = 'special',
+  VALIDATION = 'validation',
+}
+
+type Addresses = {
+  [senderId: string]: Array<DigitalAddress>;
+};
+
+const SpecialContacts: React.FC = () => {
+  const { t } = useTranslation(['common', 'recapiti']);
+  const dispatch = useAppDispatch();
+  const isMobile = useIsMobile();
+  const addressesData = useAppSelector(contactsSelectors.selectAddresses);
+  const [modalOpen, setModalOpen] = useState<ModalType | null>(null);
+
+  const currentAddress = useRef<{ value: string; sender: Sender; channelType: ChannelType }>({
+    value: '',
+    sender: { senderId: 'dafault' },
+    channelType: ChannelType.PEC,
+  });
+
+  const labelRoot =
+    currentAddress.current.channelType === ChannelType.PEC ? 'legal-contacts' : 'courtesy-contacts';
+  const contactType = currentAddress.current.channelType?.toLowerCase();
+
+  const sendSuccessEvent = (type: ChannelType) => {
+    const event =
```

Comment:
we can avoid this nested ternary operator. Try to do this:
1. define a variable `eventKey = 'SEND_ADD_${type}_UX_SUCCESS'`
2. define a function `isPFEvent` outside the component that check if the key passed as parameter is into the enumeration PFEventsType
3. use the function in this way
```
const sendSuccessEvent = (type: ChannelType) => {
    const eventKey = `SEND_ADD_${type}_UX_SUCCESS`;
    if (isPFEvent(eventKey)) {
      PFEventStrategyFactory.triggerEvent(PFEventsType[eventKey], currentAddress.current.sender.senderId);
    }
  };
```

TIPS:  for the isPFEvent use the `Object.keys` and the `includes` to check if the key is into the enumeration. The function must use the `is` constructor to work.